### PR TITLE
doc: release-notes V3.1.0: STM32 and shields updates

### PR DESCRIPTION
**Pull request by erwango**

### CODE REVIEW

File: doc/releases/release-notes-3.1.rst [hunk 1]
```diff
@@ -143,6 +143,8 @@ Bluetooth
     function has been deprecated.  A :c:func:`subscribe` callback
     function has been added to replace it.
 
+  * :c:func:`bt_disable` was added to enable caller to disable bluetooth stack.
+
```
Comment (mbolivar-nordic) [inline doc/releases/release-notes-3.1.rst:147]: Did you mean to put this in this commit?

File: doc/releases/release-notes-3.1.rst [hunk 2]
```diff
@@ -200,11 +202,16 @@ Bluetooth
 
 * HCI Driver
 
+  * Added support for a new optional c:func:`close` API which closes HCI transport.
+  * Implemented c:func:`close` on stm32wb HCI driver.
```
Comment (gmarull) [inline doc/releases/release-notes-3.1.rst:206]:
```suggestion
  * Added support for a new optional :c:func:`close` API which closes HCI transport.
  * Implemented :c:func:`close` on stm32wb HCI driver.
```

are these real functions? just use literals if not

File: doc/releases/release-notes-3.1.rst [hunk 3]
```diff
@@ -225,12 +235,17 @@ Boards & SoC Support
...
+  * Keyestudio CAN-BUS Shield (KS0411)
+  * MikroElektronika WIFI and BLE Shield
+  * X-NUCLEO-53L0A1 ranging and gesture detection sensor expansion board
 
 Drivers and Sensors
 *******************
 
 * ADC
 
+  * STM32: Added :c:macro:`ref_internal` support
```
Comment (gmarull) [inline doc/releases/release-notes-3.1.rst:247]: this macro doesn't exist

File: doc/releases/release-notes-3.1.rst (hunk 2 above)
Comment (gmarull) [reply doc/releases/release-notes-3.1.rst:206]: I think the right way to reference in this case would be:

```
:c:member:`bt_hci_driver.close`
```

File: doc/releases/release-notes-3.1.rst (hunk 3 above)
Comment (gmarull) [reply doc/releases/release-notes-3.1.rst:247]: I think it's better to do here:

```
Added support for :c:enumerator:`adc_reference:ADC_REF_INTERNAL`.
```

We do not document publicly API ops now (ADC needs to be fixed to mark that as internal indeed)

Comment (erwango) [reply doc/releases/release-notes-3.1.rst:247]: ok, thanks will do

File: doc/releases/release-notes-3.1.rst [hunk 4]
```diff
@@ -215,11 +217,17 @@ Bluetooth
 
 * HCI Driver
 
+  * Added support for a new optional :c:member:`bt_hci_driver.close` API which
+    closes HCI transport.
+  * Implemented :c:func:`close` on stm32wb HCI driver.
```
Comment (gmarull) [inline doc/releases/release-notes-3.1.rst:222]:
```suggestion
  * Implemented :c:member:`bt_hci_driver.close` on stm32wb HCI driver.
```

Comment (erwango) [reply doc/releases/release-notes-3.1.rst:222]: Oops txs

**Review by gmarull — APPROVED** [review]

File: doc/releases/release-notes-3.1.rst (hunk 1 above)
Comment (erwango) [reply doc/releases/release-notes-3.1.rst:162]: Right, split into a dedicated commit.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**